onnx/defs/schema.h

```diff
@@ -33,6 +33,21 @@ typedef std::unordered_set<DataType> DataTypeSet;
 typedef std::unordered_map<std::string, std::pair<DataTypeSet, std::string>>
     TypeConstraintMap;
 
+typedef TensorShapeProto_Dimension InferenceDimension;
+
+enum class Status {
```

Comment:
why status - we can use exceptions for error propagation (as the rest of schema checking does)

Reply:
I have no preference, @linkerzhang requested it. @linkerzhang are you ok with switching back to void return and using exceptions for errors?

Reply:
Rest of ONNX code uses exceptions (e.g. checker) - so unless there are strong concerns I'd prefer that, for consistency

Reply:
I was planning to introduce a "status" class into ONNX to avoid exceptions, but haven't gotten chance to do that. I'd suggest to move to the direction, although it takes time to change all others to return status, I can do that later.